core/src/main/java/org/elasticsearch/client/transport/TransportClientNodesService.java

```diff
@@ -378,11 +380,13 @@ public LivenessResponse newInstance() {
                         logger.warn("node {} not part of the cluster {}, ignoring...", listedNode, clusterName);
                         newFilteredNodes.add(listedNode);
                     } else if (livenessResponse.getDiscoveryNode() != null) {
-                        // use discovered information but do keep the original transport address, so people can control which address is exactly used.
+                        // use discovered information but do keep the original transport address,
+                        // so people can control which address is exactly used.
                         DiscoveryNode nodeWithInfo = livenessResponse.getDiscoveryNode();
-                        newNodes.add(new DiscoveryNode(nodeWithInfo.name(), nodeWithInfo.id(), nodeWithInfo.getHostName(), nodeWithInfo.getHostAddress(), listedNode.address(), nodeWithInfo.attributes(), nodeWithInfo.version()));
+                        newNodes.add(new DiscoveryNode(nodeWithInfo.name(), nodeWithInfo.id(), nodeWithInfo));
```

Comment:
yes we did, thanks a lot for catching it, I fixed and added a test for this. Mind having a look at it? I don't personally love it but couldn't find a better way to test it...